pkg/testsuite/benchmarks_test.go

```diff
@@ -0,0 +1,578 @@
+	fmt.Printf("Output: %d\n", normalisedCoeffs[3])
+	fmt.Printf("NativeToken: %d\n", normalisedCoeffs[4])
+	fmt.Printf("Staking: %d\n", normalisedCoeffs[5])
+	fmt.Printf("BlockIssuer: %d\n", normalisedCoeffs[6])
+	fmt.Printf("Allotment: %d\n", normalisedCoeffs[7])
+	fmt.Printf("SignatureEd25519: %d\n", normalisedCoeffs[8])
+}
+
+func getBlockWorkScoreRegressors(block *iotago.Block) []float64 {
+	regressors := make([]float64, 8)
+
+	basicBlockBody, isBasic := block.Body.(*iotago.BasicBlockBody)
+	if !isBasic {
+		panic("block body is not a basic block body")
+	}
+	signedTx, isSignedTx := basicBlockBody.Payload.(*iotago.SignedTransaction)
+	if !isSignedTx {
+		panic("block payload is not a signed transaction")
+	}
+	// add one to the Input regressor for each input
+	regressors[0] += float64(len(signedTx.Transaction.TransactionEssence.Inputs))
+	// add one to the ContextInput regressor for each context input
+	regressors[1] += float64(len(signedTx.Transaction.TransactionEssence.ContextInputs))
+	for _, output := range signedTx.Transaction.Outputs {
+		// add one to the Output regressor for each output
+		regressors[2] += 1
+		for _, feature := range output.FeatureSet() {
+			switch feature.Type() {
+			case iotago.FeatureNativeToken:
+				// add one to the NativeToken regressor for each output with the native token feature
+				regressors[3] += 1
+			case iotago.FeatureStaking:
+				// add one to the Staking regressor for each output with the staking feature
+				regressors[4] += 1
+			case iotago.FeatureBlockIssuer:
+				// add one to the BlockIssuer regressor for each output with the block issuer feature
+				regressors[5] += 1
+			}
+		}
+	}
+	// add one to Allotments regressor for each allotment
+	regressors[6] += float64(len(signedTx.Transaction.TransactionEssence.Allotments))
+	for _, unlock := range signedTx.Unlocks {
+		if unlock.Type() == iotago.UnlockSignature {
+			// add one to the SignatureEd25519 regressor for each unlock block
+			regressors[7] += 1
+		}
+	}
+
+	return regressors
+}
+
+// workScoreParamsFromCoefficients creates a WorkScoreParameters from the coefficients provided by the regression model.
+// The parameters are scaled such that the the standardBlock has workScore of standardBlockMinCost. This represents the Mana cost
+// of the "standard block" when the reference Mana cost (RMC) is at its minimum value of 1.
+// The dataByteRatio is the ratio of the part of the WorkScore due to the DataBytes factor to that due to the rest of the factors.
+func workScoreParamsFromCoefficients(coeffs []float64, standardBlock *iotago.Block, standardBlockMinCost iotago.Mana, dataByteRatio float64) iotago.WorkScoreParameters {
+	standarBlockRegressors := getBlockWorkScoreRegressors(standardBlock)
+	fmt.Printf("Standard block regressors: %+v\n", standarBlockRegressors)
+	standardBlockWorkScore := coeffs[0]
+	for i, regressor := range standarBlockRegressors {
+		standardBlockWorkScore += regressor * coeffs[i+1]
+	}
```

Comment:
Can you add some comments there about the indexes? I don't get why it is "i+1" on the first view.